dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -1328,10 +1256,161 @@ private Response response ( String response, Boolean error ) {
+                            try {
+                                integrityUtil.dropTempTables(endpoint.getId());
+                                HibernateUtil.closeSession();
+                            } catch (DotHibernateException e) {
+                                Logger.warn(this, e.getMessage(), e);
+                            } catch (DotDataException e) {
+                                Logger.error(IntegrityResource.class, "Error while deleting temp tables", e);
+                            }
+                        }
+
+                        if (conflictPresent) {
+                            //Setting the process status
+                            setStatus(session, endpoint.getId(), ProcessStatus.FINISHED, null);
+                        } else {
+                            String noConflictMessage;
+                            try {
+                                noConflictMessage = LanguageUtil.get(loggedUser.getLocale(), "push_publish_integrity_conflicts_not_found");
+                            } catch (LanguageException e) {
+                                noConflictMessage = "No Integrity Conflicts found";
+                            }
+                            //Setting the process status
+                            setStatus(session, endpoint.getId(), ProcessStatus.NO_CONFLICTS, noConflictMessage);
+                        }
+
+                    } else if (response.getStatus() == HttpStatus.SC_PROCESSING) {
+
+                        continue;
+                    } else if (response.getStatus() == HttpStatus.SC_RESET_CONTENT) {
+                        processing = false;
+                        //Setting the process status
+                        setStatus(session, endpoint.getId(), ProcessStatus.CANCELED, null);
+                    } else {
+                        setStatus(session, endpoint.getId(), ProcessStatus.ERROR, null);
+                        Logger.error(this.getClass(), "Response indicating a " + response.getStatusInfo().getReasonPhrase() + " (" + response.getStatus() + ") Error trying to retrieve the Integrity data from the Endpoint [" + endpoint.getId() + "].");
+                        processing = false;
+                    }
+                } catch (NotEndPointTokenFoundException e) {
+                    Logger.warn(IntegrityResource.class, "No Auth Token set for endpoint");
+                    setStatus(session, endpoint.getId(), ProcessStatus.ERROR, null);
+                }
+            }
+        }
+
+        @NotNull
+        private Response statusIntegrityCheckerRequest() throws NotEndPointTokenFoundException {
+            String url = String.format("%s/api/integrity/%s/status", endpoint.toURL(), integrityDataRequestID);
+
+            return postWithEndpointState(
+                    endpoint, url, new MediaType("application", "zip"));
+        }
+    }
+
+    private static class NotEndPointTokenFoundException extends Throwable {
```

Comment:
Codacy found an issue: [Exceptions should not extend java.lang.Throwable](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)